pkg/odo/cli/component/push.go

```diff
@@ -40,10 +40,10 @@ type PushOptions struct {
 	DevfilePath string
 
 	// devfile commands
+	devfileInitCommand  string
```

Comment:
This command was added to push.go and it is also accepting a custom command in function `GetInitCommand()` but `odo push` does not have a new flag to pass the custom init command from the command line.. Users can also push with a devfile without needing to always have a `devinit` command.. This need to be fixed before being merged.